modules/ref_fb_module/src/scaling_fb_impl.cpp

```diff
@@ -251,11 +251,17 @@ void ScalingFbImpl::createSignals()
 
 void ScalingFbImpl::initStatuses()
 {
-    if (!this->context.getTypeManager().hasType("InputStatusType"))
+    auto inputStatusType = EnumerationType("InputStatusType", List<IString>(InputDisconnected, InputConnected, InputInvalid));
+
+    try
     {
-        auto inputStatusType = EnumerationType("InputStatusType", List<IString>(InputDisconnected, InputConnected, InputInvalid));
         this->context.getTypeManager().addType(inputStatusType);
     }
+    catch (const std::exception& e)
```

Comment:
In a lot of these catch statements I'd format them as:
try
{}
catch(const std::exception& e)
{log with error info}
catch(...)
{log without error info}